Add Atom package from bigtest

jnicklas opened the pull request:
This imports the atom package from BigTest into the repo and makes it work with v2 effection. The ability to reset an atom has been removed.

cowboyd commented:
Funny, looking at it now, if we aliased the `set()` method to `send()`, `Slice` would also be a `WritableStream` 🤔

cowboyd commented on packages/atom/src/types.ts:
```diff
@@ -0,0 +1,112 @@
+import { Stream } from '@effection/subscription';
```
Even though this is a core package, should we import from `effection`? Feels like when we deprecate subscription, it'll be one less point of coupling. However, I can also see that in order to avoid circular dependencies we could have a policy that core package depend on core packages explicitly.

Or maybe the rule is "blessed packages should depend directly on other blessed packages", which in this case, means that atom could pull from the omnibus `effection` package since it is not a blessed package (and is unlikely to be one)

jnicklas replied:
Actually in this PR I did add `atom` as a blessed package, but maybe it shouldn't be?

IMO all packages in the monorepo should depend on the individual subpackages, regardless of whether they are blessed or not.

cowboyd replied:
I'd say that this is not a blessed package as you will only need it a certain amount of time that is likely to be < 50%

👍🏻 on the policy of importing from individual packages within the repo.

jnicklas commented:
@cowboyd is this good to go? Should `atom` be a blessed package?

cowboyd commented:
I think this is pretty much good to go, but I think atom should _not_ be a blessed package.

jnicklas committed:
Add Atom package from bigtest

This imports the atom package from BigTest into the repo and makes it work with v2 effection. The ability to reset an atom has been removed.

jnicklas commented:
@cowboyd I removed it as blessed package

cowboyd reviewed (approved):
😍
